File: MiniDOMOffice/Tickets/Messages/CTicketAnsware.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;

namespace minidom
{
    public partial class Office
    {


        /// <summary>
        /// Rappresenta una richiesta fatta da un utente
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class CTicketAnsware 
            : minidom.Databases.DBObject, IComparable, IComparable<CTicketAnsware>
        {
            private int m_IDTicket;
            [NonSerialized] private CTicket m_Ticket;
            private int m_IDOperatore;
            [NonSerialized] private CUser m_Operatore;
            private string m_NomeOperatore;
            private DateTime m_Data;
            private TicketStatus m_StatoTicket;
            private string m_Messaggio;
            private CCollection<CAttachment> m_Attachments;

            /// <summary>
            /// Costruttore
            /// </summary>
            public CTicketAnsware()
            {
                m_IDTicket = 0;
                m_Ticket = null;
                m_IDOperatore = 0;
                m_Operatore = null;
                m_NomeOperatore = "";
                m_Data = DMD.DateUtils.Now();
                m_Messaggio = "";
                m_Attachments = null;
            }

            /// <summary>
            /// Restituisce o imposta l'ID del ticket a cui appartiene il messaggio
            /// </summary>
            /// <value></value>
            /// <returns></returns>
            /// <remarks></remarks>
            public int IDTicket
            {
                get
                {
                    return DBUtils.GetID(m_Ticket, m_IDTicket);
                }

                set
                {
                    int oldValue = IDTicket;
                    if (oldValue == value)
                        return;
                    m_IDTicket = value;
                    m_Ticket = null;
                    DoChanged("Ticket", value, oldValue);
                }
            }


            /// <summary>
            /// Restituisce o imposta il ticket a cui appartiene il messaggio
            /// </summary>
            /// <value></value>
            /// <returns></returns>
            /// <remarks></remarks>
            public CTicket Ticket
            {
                get
                {
                    lock (this)
                    {
                        if (m_Ticket is null)
                            m_Ticket = Tickets.GetItemById(m_IDTicket);
                        return m_Ticket;
                    }
                }

                set
                {
                    CTicket oldValue;
                    lock (this)
                    {
                        oldValue = m_Ticket;
                        if (ReferenceEquals(oldValue, value))
                            return;
                        m_Ticket = value;
                        m_IDTicket = DBUtils.GetID(value, 0);
                    }

                    DoChanged("Ticket", value, oldValue);
                }
            }

            /// <summary>
            /// Imposta il ticket
            /// </summary>
            /// <param name="value"></param>
            protected internal void SetTicket(CTicket value)
            {
                m_Ticket = value;
                m_IDTicket = DBUtils.GetID(value, 0);
            }

            /// <summary>
            /// Restituisce o imposta l'ID dell'operatore che ha inserito il messaggio
            /// </summary>
            /// <value></value>
            /// <returns></returns>
            /// <remarks></remarks>
            public int IDOperatore
            {
                get
                {
                    return DBUtils.GetID(m_Operatore, m_IDOperatore);
                }

                set
                {
                    int oldValue = IDOperatore;
                    if (oldValue == value)
                        return;
                    m_IDOperatore = value;
                    m_Operatore = null;
                    DoChanged("IDOperatore", value, oldValue);
                }
            }

            /// <summary>
            /// Restituisce o imposta l'operatore che ha inserito il messaggio
            /// </summary>
            /// <value></value>
            /// <returns></returns>
            /// <remarks></remarks>
            public CUser Operatore
            {
                get
                {
                    lock (this)
                    {
                        if (m_Operatore is null)
                            m_Operatore = Sistema.Users.GetItemById(m_IDOperatore);
                        return m_Operatore;
                    }
                }

                set
                {
                    Sistema.CUser oldValue;
                    lock (this)
                    {
                        oldValue = Operatore;
                        if (ReferenceEquals(oldValue, value))
                            return;
                        m_Operatore = value;
                        m_IDOperatore = DBUtils.GetID(value, 0);
                        m_NomeOperatore = (value is object)? value.Nominativo : "";
                    }

                    DoChanged("Operatore", value, oldValue);
                }
            }

            /// <summary>
            /// Restituisce o imposta il nome dell'operatore che ha inserito il messaggio
            /// </summary>
            /// <value></value>
            /// <returns></returns>
            /// <remarks></remarks>
            public string NomeOperatore
            {
                get
                {
                    return m_NomeOperatore;
                }

                set
                {
                    string oldValue = m_NomeOperatore;
                    value = DMD.Strings.Trim(value);
                    if ((oldValue ?? "") == (value ?? ""))
                        return;
                    m_NomeOperatore = value;
                    DoChanged("NomeOperatore", value, oldValue);
                }
            }

            /// <summary>
            /// Restituisce o imposta la data in cui è stato inserito il messaggio
            /// </summary>
            /// <value></value>
            /// <returns></returns>
            /// <remarks></remarks>
            public DateTime Data
            {
                get
                {
                    return m_Data;
                }

                set
                {
                    var oldValue = m_Data;
                    if (DMD.DateUtils.Compare(value, oldValue) == 0)
                        return;
                    m_Data = value;
                    DoChanged("Data", value, oldValue);
                }
            }

            /// <summary>
            /// Restituisce o imposta il messaggio
            /// </summary>
            /// <value></value>
            /// <returns></returns>
            /// <remarks></remarks>
            public string Messaggio
            {
                get
                {
                    return m_Messaggio;
                }

                set
                {
                    string oldValue = m_Messaggio;
                    if ((oldValue ?? "") == (value ?? ""))
                        return;
                    m_Messaggio = value;
                    DoChanged("Messaggio", value, oldValue);
                }
            }

            /// <summary>
            /// Restituisce o imposta lo stato in cui viene portato il ticket in seguito a questa risposta
            /// </summary>
            /// <value></value>
            /// <returns></returns>
            /// <remarks></remarks>
            public TicketStatus StatoTicket
            {
                get
                {
                    return m_StatoTicket;
                }

                set
                {
                    var oldValue = m_StatoTicket;
                    if (oldValue == value)
                        return;
                    m_StatoTicket = value;
                    DoChanged("StatoTicket", value, oldValue);
                }
            }

            /// <summary>
            /// Allegati
            /// </summary>
            public CCollection<CAttachment> Attachments
            {
                get
                {
                    lock (this)
                    {
                        if (m_Attachments is null)
                            m_Attachments = new CCollection<CAttachment>();
                        return m_Attachments;
                    }
                }
            }

            /// <summary>
            /// Restituisce una stringa che rappresenta l'oggetto
            /// </summary>
            /// <returns></returns>
            public override string ToString()
            {
                return m_Messaggio;
            }

            /// <summary>
            /// Restituisce il codice hash dell'oggetto
            /// </summary>
            /// <returns></returns>
            public override int GetHashCode()
            {
                return this.m_Data.GetHashCode();
            }

            /// <summary>
            /// Restituisce true se i due oggetti sono uguali
            /// </summary>
            /// <param name="obj"></param>
            /// <returns></returns>
            public override bool Equals(Databases.DBObject obj)
            {
                return (obj is CTicketAnsware) && this.Equals((CTicketAnsware)obj);
            }

            /// <summary>
            /// Restituisce true se i due oggetti sono uguali
            /// </summary>
            /// <param name="obj"></param>
            /// <returns></returns>
            public virtual bool Equals(CTicketAnsware obj)
            {
                return base.Equals(obj)
                    && DMD.Integers.EQ(this.m_IDTicket, obj.m_IDTicket)
                    && DMD.Integers.EQ(this.m_IDOperatore, obj.m_IDOperatore)
                    && DMD.Strings.EQ(this.m_NomeOperatore, obj.m_NomeOperatore)
                    && DMD.DateUtils.EQ(this.m_Data, obj.m_Data)
                    && DMD.Integers.EQ((int)this.m_StatoTicket, (int)obj.m_StatoTicket)
                    && DMD.Strings.EQ(this.m_Messaggio, obj.m_Messaggio)
                    ;
                    //private CCollection<CAttachment> m_Attachments;
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            public override CModulesClass GetModule()
            {
                return minidom.Office.Tickets.Messages;
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            public override string GetTableName()
            {
                return "tbl_SupportTicketsAnswares";
            }

            /// <summary>
            /// Carica dal db
            /// </summary>
            /// <param name="reader"></param>
            /// <returns></returns>
            protected override bool LoadFromRecordset(DBReader reader)
            {
                this.m_IDTicket = reader.Read("IDTicket", this.m_IDTicket);
                this.m_IDOperatore = reader.Read("IDOperatore", this.m_IDOperatore);
                this.m_NomeOperatore = reader.Read("NomeOperatore", this.m_NomeOperatore);
                this.m_Data = reader.Read("Data", this.m_Data);
                this.m_Messaggio = reader.Read("Messaggio", this.m_Messaggio);
                this.m_StatoTicket = reader.Read("StatoTicket", this.m_StatoTicket);
                string tmp = reader.Read("Attachments", "");
                if (!string.IsNullOrEmpty(tmp))
                {
                    var col = (CCollection)DMD.XML.Utils.Serializer.Deserialize(tmp);
                    m_Attachments = new CCollection<CAttachment>();
                    m_Attachments.AddRange(col);
                }

                return base.LoadFromRecordset(reader);
            }

            /// <summary>
            /// Salva nel db
            /// </summary>
            /// <param name="writer"></param>
            /// <returns></returns>
            protected override bool SaveToRecordset(DBWriter writer)
            {
                writer.Write("IDTicket", this.IDTicket);
                writer.Write("IDOperatore", this.IDOperatore);
                writer.Write("NomeOperatore", this.m_NomeOperatore);
                writer.Write("Data", this.m_Data);
                writer.Write("Messaggio", this.m_Messaggio);
                writer.Write("StatoTicket", this.m_StatoTicket);
                writer.Write("Attachments", DMD.XML.Utils.Serializer.Serialize(this.Attachments));
                return base.SaveToRecordset(writer);
            }

            /// <summary>
            /// Prepara i campi
            /// </summary>
            /// <param name="table"></param>
            protected override void PrepareDBSchemaFields(DBTable table)
            {
                base.PrepareDBSchemaFields(table);

                var c = table.Fields.Ensure("IDTicket", typeof(int), 1);
                c = table.Fields.Ensure("IDOperatore", typeof(int), 1);
                c = table.Fields.Ensure("NomeOperatore", typeof(string), 255);
                c = table.Fields.Ensure("Data", typeof(DateTime), 1);
                c = table.Fields.Ensure("Messaggio", typeof(string), 0);
                c = table.Fields.Ensure("StatoTicket", typeof(int), 1);
                c = table.Fields.Ensure("Attachments", typeof(string), 0);
            }

            /// <summary>
            /// Prepara i vincoli
            /// </summary>
            /// <param name="table"></param>
            protected override void PrepareDBSchemaConstraints(DBTable table)
            {
                base.PrepareDBSchemaConstraints(table);

                var c = table.Constraints.Ensure("idxTicket", new string[] { "IDTicket" }, DBFieldConstraintFlags.None);
                c = table.Constraints.Ensure("idxOperatore", new string[] { "IDOperatore", "NomeOperatore" }, DBFieldConstraintFlags.None);
                c = table.Constraints.Ensure("idxData", new string[] { "Data" }, DBFieldConstraintFlags.None);
                c = table.Constraints.Ensure("idxMessaggio", new string[] { "Messaggio" }, DBFieldConstraintFlags.None);
                c = table.Constraints.Ensure("idStatoTicket", new string[] { "StatoTicket" }, DBFieldConstraintFlags.None);
                //c = table.Fields.Ensure("Attachments", typeof(string), 0);
            }

            /// <summary>
            /// Deserializzazione xml
            /// </summary>
            /// <param name="writer"></param>
            protected override void XMLSerialize(XMLWriter writer)
            {
                writer.WriteAttribute("IDTicket", this.IDTicket);
                writer.WriteAttribute("IDOperatore", this.IDOperatore);
                writer.WriteAttribute("NomeOperatore", this.m_NomeOperatore);
                writer.WriteAttribute("Data", this.m_Data);
                writer.WriteAttribute("StatoTicket", (int?)this.m_StatoTicket);
                base.XMLSerialize(writer);
                writer.WriteTag("Attachments", this.Attachments);
                writer.WriteTag("Messaggio", this.m_Messaggio);
            }

            /// <summary>
            /// Deserializzazione xml
            /// </summary>
            /// <param name="fieldName"></param>
            /// <param name="fieldValue"></param>
            protected override void SetFieldInternal(string fieldName, object fieldValue)
            {
                switch (fieldName ?? "")
                {
                    case "IDTicket":
                        {
                            m_IDTicket = (int)DMD.XML.Utils.Serializer.DeserializeInteger(fieldValue);
                            break;
                        }

                    case "IDOperatore":
                        {
                            m_IDOperatore = (int)DMD.XML.Utils.Serializer.DeserializeInteger(fieldValue);
                            break;
                        }

                    case "NomeOperatore":
                        {
                            m_NomeOperatore = DMD.XML.Utils.Serializer.DeserializeString(fieldValue);
                            break;
                        }

                    case "Data":
                        {
                            m_Data = (DateTime)DMD.XML.Utils.Serializer.DeserializeDate(fieldValue);
                            break;
                        }

                    case "StatoTicket":
                        {
                            m_StatoTicket = (TicketStatus)DMD.XML.Utils.Serializer.DeserializeInteger(fieldValue);
                            break;
                        }

                    case "Messaggio":
                        {
                            m_Messaggio = DMD.XML.Utils.Serializer.DeserializeString(fieldValue);
                            break;
                        }

                    case "Attachments":
                        {
                            m_Attachments = new CCollection<Sistema.CAttachment>();
                            m_Attachments.AddRange((IEnumerable)fieldValue);
                            break;
                        }

                    default:
                        {
                            base.SetFieldInternal(fieldName, fieldValue);
                            break;
                        }
                }
            }

            /// <summary>
            /// Compara due messaggi per data
            /// </summary>
            /// <param name="other"></param>
            /// <returns></returns>
            public int CompareTo(CTicketAnsware other)
            {
                return DMD.DateUtils.Compare(m_Data, other.m_Data);
            }

            int IComparable.CompareTo(object obj)
            {
                return CompareTo((CTicketAnsware)obj);
            }
        }
    }
}

File: MiniDOMOffice/Tickets/Messages/CTicketAnsware.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica

Partial Public Class Office

     
    ''' <summary>
    ''' Rappresenta una richiesta fatta da un utente
    ''' </summary>
    ''' <remarks></remarks>
    <Serializable> _
    Public NotInheritable Class CTicketAnsware
        Inherits DBObject
        Implements IComparable

        Private m_IDTicket As Integer
        <NonSerialized> Private m_Ticket As CTicket
        Private m_IDOperatore As Integer
        <NonSerialized> Private m_Operatore As CUser
        Private m_NomeOperatore As String
        Private m_Data As Date
        Private m_StatoTicket As TicketStatus
        Private m_Messaggio As String
        Private m_Attachments As CCollection(Of CAttachment)

        Public Sub New()
            Me.m_IDTicket = 0
            Me.m_Ticket = Nothing
            Me.m_IDOperatore = 0
            Me.m_Operatore = Nothing
            Me.m_NomeOperatore = ""
            Me.m_Data = DateUtils.Now
            Me.m_Messaggio = ""
            Me.m_Attachments = Nothing
        End Sub

        ''' <summary>
        ''' Restituisce o imposta l'ID del ticket a cui appartiene il messaggio
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Property IDTicket As Integer
            Get
                Return GetID(Me.m_Ticket, Me.m_IDTicket)
            End Get
            Set(value As Integer)
                Dim oldValue As Integer = Me.IDTicket
                If (oldValue = value) Then Exit Property
                Me.m_IDTicket = value
                Me.m_Ticket = Nothing
                Me.DoChanged("Ticket", value, oldValue)
            End Set
        End Property


        ''' <summary>
        ''' Restituisce o imposta il ticket a cui appartiene il messaggio
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Property Ticket As CTicket
            Get
                SyncLock Me
                    If (Me.m_Ticket Is Nothing) Then Me.m_Ticket = Tickets.GetItemById(Me.m_IDTicket)
                    Return Me.m_Ticket
                End SyncLock
            End Get
            Set(value As CTicket)
                Dim oldValue As CTicket
                SyncLock Me
                    oldValue = Me.m_Ticket
                    If (oldValue Is value) Then Exit Property
                    Me.m_Ticket = value
                    Me.m_IDTicket = GetID(value)
                End SyncLock
                Me.DoChanged("Ticket", value, oldValue)
            End Set
        End Property

        Protected Friend Sub SetTicket(ByVal value As CTicket)
            Me.m_Ticket = value
            Me.m_IDTicket = GetID(value)
        End Sub

        ''' <summary>
        ''' Restituisce o imposta l'ID dell'operatore che ha inserito il messaggio
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Property IDOperatore As Integer
            Get
                Return GetID(Me.m_Operatore, Me.m_IDOperatore)
            End Get
            Set(value As Integer)
                Dim oldValue As Integer = Me.IDOperatore
                If (oldValue = value) Then Exit Property
                Me.m_IDOperatore = value
                Me.m_Operatore = Nothing
                Me.DoChanged("IDOperatore", value, oldValue)
            End Set
        End Property

        ''' <summary>
        ''' Restituisce o imposta l'operatore che ha inserito il messaggio
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Property Operatore As CUser
            Get
                SyncLock Me
                    If Me.m_Operatore Is Nothing Then Me.m_Operatore = Sistema.Users.GetItemById(Me.m_IDOperatore)
                    Return Me.m_Operatore
                End SyncLock
            End Get
            Set(value As CUser)
                Dim oldValue As CUser
                SyncLock Me
                    oldValue = Me.Operatore
                    If (oldValue Is value) Then Exit Property
                    Me.m_Operatore = value
                    Me.m_IDOperatore = GetID(value)
                    If (value IsNot Nothing) Then Me.m_NomeOperatore = value.Nominativo
                End SyncLock
                Me.DoChanged("Operatore", value, oldValue)
            End Set
        End Property

        ''' <summary>
        ''' Restituisce o imposta il nome dell'operatore che ha inserito il messaggio
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Property NomeOperatore As String
            Get
                Return Me.m_NomeOperatore
            End Get
            Set(value As String)
                Dim oldValue As String = Me.m_NomeOperatore
                value = Strings.Trim(value)
                If (oldValue = value) Then Exit Property
                Me.m_NomeOperatore = value
                Me.DoChanged("NomeOperatore", value, oldValue)
            End Set
        End Property

        ''' <summary>
        ''' Restituisce o imposta la data in cui è stato inserito il messaggio
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Property Data As Date
            Get
                Return Me.m_Data
            End Get
            Set(value As Date)
                Dim oldValue As Date = Me.m_Data
                If (DateUtils.Compare(value, oldValue) = 0) Then Exit Property
                Me.m_Data = value
                Me.DoChanged("Data", value, oldValue)
            End Set
        End Property

        ''' <summary>
        ''' Restituisce o imposta il messaggio
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Property Messaggio As String
            Get
                Return Me.m_Messaggio
            End Get
            Set(value As String)
                Dim oldValue As String = Me.m_Messaggio
                If (oldValue = value) Then Exit Property
                Me.m_Messaggio = value
                Me.DoChanged("Messaggio", value, oldValue)
            End Set
        End Property

        ''' <summary>
        ''' Restituisce o imposta lo stato in cui viene portato il ticket in seguito a questa risposta
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Property StatoTicket As TicketStatus
            Get
                Return Me.m_StatoTicket
            End Get
            Set(value As TicketStatus)
                Dim oldValue As TicketStatus = Me.m_StatoTicket
                If (oldValue = value) Then Exit Property
                Me.m_StatoTicket = value
                Me.DoChanged("StatoTicket", value, oldValue)
            End Set
        End Property

        Public ReadOnly Property Attachments As CCollection(Of CAttachment)
            Get
                SyncLock Me
                    If Me.m_Attachments Is Nothing Then Me.m_Attachments = New CCollection(Of CAttachment)
                    Return Me.m_Attachments
                End SyncLock
            End Get
        End Property


        Public Overrides Function ToString() As String
            Return Me.m_Messaggio
        End Function


        Protected Overrides Function GetConnection() As CDBConnection
            Return Office.Database
        End Function

        Public Overrides Function GetModule() As CModule
            Return Nothing
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_SupportTicketsAnswares"
        End Function

        Protected Overrides Function LoadFromRecordset(reader As DBReader) As Boolean
            Dim str As String
            Dim tmp As CCollection

            Me.m_IDTicket = reader.Read("IDTicket", Me.m_IDTicket)
            Me.m_IDOperatore = reader.Read("IDOperatore", Me.m_IDOperatore)
            Me.m_NomeOperatore = reader.Read("NomeOperatore", Me.m_NomeOperatore)
            Me.m_Data = reader.Read("Data", Me.m_Data)
            Me.m_Messaggio = reader.Read("Messaggio", Me.m_Messaggio)
            Me.m_StatoTicket = reader.Read("StatoTicket", Me.m_StatoTicket)
            str = reader.Read("Attachments", "")
            If (str <> "") Then
                tmp = XML.Utils.Serializer.Deserialize(str)
                Me.m_Attachments = New CCollection(Of CAttachment)
                Me.m_Attachments.AddRange(tmp)
            End If

            Return MyBase.LoadFromRecordset(reader)
        End Function

        Protected Overrides Function SaveToRecordset(writer As DBWriter) As Boolean
            writer.Write("IDTicket", Me.IDTicket)
            writer.Write("IDOperatore", Me.IDOperatore)
            writer.Write("NomeOperatore", Me.m_NomeOperatore)
            writer.Write("Data", Me.m_Data)
            writer.Write("Messaggio", Me.m_Messaggio)
            writer.Write("StatoTicket", Me.m_StatoTicket)
            writer.Write("Attachments", XML.Utils.Serializer.Serialize(Me.Attachments))
            Return MyBase.SaveToRecordset(writer)
        End Function

        Protected Overrides Sub XMLSerialize(writer As XML.XMLWriter)
            writer.WriteAttribute("IDTicket", Me.IDTicket)
            writer.WriteAttribute("IDOperatore", Me.IDOperatore)
            writer.WriteAttribute("NomeOperatore", Me.m_NomeOperatore)
            writer.WriteAttribute("Data", Me.m_Data)
            writer.WriteAttribute("StatoTicket", Me.m_StatoTicket)
            MyBase.XMLSerialize(writer)
            writer.WriteTag("Attachments", Me.Attachments)
            writer.WriteTag("Messaggio", Me.m_Messaggio)
        End Sub

        Protected Overrides Sub SetFieldInternal(fieldName As String, fieldValue As Object)
            Select Case fieldName
                Case "IDTicket" : Me.m_IDTicket = XML.Utils.Serializer.DeserializeInteger(fieldValue)
                Case "IDOperatore" : Me.m_IDOperatore = XML.Utils.Serializer.DeserializeInteger(fieldValue)
                Case "NomeOperatore" : Me.m_NomeOperatore = XML.Utils.Serializer.DeserializeString(fieldValue)
                Case "Data" : Me.m_Data = XML.Utils.Serializer.DeserializeDate(fieldValue)
                Case "StatoTicket" : Me.m_StatoTicket = XML.Utils.Serializer.DeserializeInteger(fieldValue)
                Case "Messaggio" : Me.m_Messaggio = XML.Utils.Serializer.DeserializeString(fieldValue)
                Case "Attachments" : Me.m_Attachments = New CCollection(Of CAttachment) : Me.m_Attachments.AddRange(fieldValue)
                Case Else : MyBase.SetFieldInternal(fieldName, fieldValue)
            End Select
        End Sub

        Public Function CompareTo(ByVal other As CTicketAnsware) As Integer
            Return DateUtils.Compare(Me.m_Data, other.m_Data)
        End Function
 
        Private Function CompareTo1(obj As Object) As Integer Implements IComparable.CompareTo
            Return Me.CompareTo(CType(obj, CTicketAnsware))
        End Function
    End Class


End Class